### ISSUE TITLE
Nearest road stuck in index: missing 146, Tolvajos tanya, Orgovány

### ISSUE THREAD

**Issue by bkil** [osm-search/Nominatim#2140]
## What did you search for?

- https://nominatim.openstreetmap.org/ui/search.html?street=146+Tolvajos+tanya&city=Orgov%C3%A1ny

(It doesn't matter whether you add Hungary or not)

## What result did you get?

0 results.

## What result did you expect?

- https://nominatim.openstreetmap.org/details.php?osmtype=W&osmid=844314129&format=json
- https://nominatim.openstreetmap.org/ui/details.html?osmtype=W&osmid=844314129

**Is the result in the right place and just named wrongly?** 

No, the result list is empty.

**Is the result missing completely?**

Yes.

## Further details

Inspecting the above Nominatim details page of the object, it can be seen that it is associated with the wrong address. The point seems to be linked to a nearby road that is named differently. If you search for the wrong address, the above object is indeed returned:

- https://nominatim.openstreetmap.org/ui/search.html?city=Orgov%C3%A1ny&street=146%20L%C5%91t%C3%A9ri%20%C3%BAt

### A little more background

As can be seen from the history of the building, many addresses around there were retagged from addr:place=* to addr:street=* for the very reason so that Nominatim will find them. This does work in most cases, but somehow after multiple tag changes here (most recently on 2020-11-22), the Nominatim index seems to have been corrupted in a way that the last index refresh on 2020-12-07 did not clear up the false association between the closest road and the point.

- https://www.openstreetmap.org/way/844314129/history#map=18/46.72909/19.45478

After checking another specimen that incidentally can be found successfully using Nominatim, it can be seen from the details page that it too is associated to a street that is by chance called the same as the street of the building, so it probably only works by accident right now:
- https://nominatim.openstreetmap.org/ui/details.html?osmtype=W&osmid=851419023

This was only an example, I think many more buildings are affected by this.

**Comment by lonvia** [comment]
I can't find the street 'Tolvajos tanya' that the address refers to. Could you give an OSM link to it?

**Comment by bkil** [comment]
A [tanya](https://en.wikipedia.org/wiki/Sala%C5%A1) in this context is only considered a pseudo-street. Originally each `tanya` was a single huge farm, but then later on they started to split it together to smaller allotments and started numbering them.

Usually no such street is officially designated or known. It's just an area where houses are grouped together. If you do see similarly named street nearby, they may have been just attempts at working around this issue but they are usually (but not always) unofficial. I think in most cases, one could synthesize their extent using areas or relations.

Can Nominatim only find addresses for which street ways (or areas and relations) exist? This doesn't seem universal around the world, but at least it could be documented. The resulting behavior is against the principle of least surprise at the least.

**Comment by lonvia** [comment]
If it is a farm like object than `addr:place` was the correct way to tag it. It was exactly made for house numbers that do not refer to a street, see https://wiki.openstreetmap.org/wiki/Key:addr:place. Revert your edits and Nominatim will do find the address as expected. I'd probably add a `place=isolated_dwelling` or `place=hamlet` with `name=Tolvajos tanya` although its not mandatory for search.

**Comment by bkil** [comment]
* Nope, the identifier of the dwelling isn't `Tolvajos tanya`, it's `Tolvajos tanya 146`. You can see Tolvajos tanya 147 and [Tolvajos tanya 129](https://www.openstreetmap.org/way/848755783) nearby.
* It was not my edit, but we have a country-wide agreement that we tag "tanya" like this:
  *  https://wiki.openstreetmap.org/wiki/Hungary/Jel%C3%B6l%C3%A9si_p%C3%A9ld%C3%A1k

**Comment by bkil** [comment]
I see that #912 is linked from the wiki page. Although, it seems to refer to countries where addr:place=* contains the village name. I don't think that moving back to addr:place=* (for this particular data consumer) and creating pseudo-points for `Tolvajos tanya` (and every other tanya) at a random place would be a feasible solution. Creating pseudo-streets out of tracks would make more sense here, but I still don't see what is the rationale behind this restriction. Is it so that Nominatim can interpolate the position of unknown house numbers?

**Comment by lonvia** [comment]
I did not say that the identifier of the dwelling is `Tolvajos tanya`. I said that the house number of the house does not refer to a street but to a locality. And that seems to be exactly the situation that you describe. Now, if there is no actual locality named `Tolvajos tanya` then don't add a place node. You still need to use `addr:place` to indicate that it is not a house number is not attached to a street. This is a common agreement that is in use all over the world and Nominatim will not implement an exception for Hungary.

The Wiki page you cite has it backwards with "Magyarországi gyakorlat, hogy használjuk az addr:street=*-et erre a célra is, különben a Nominatim nem találja meg." Nominatim will not find it when you use `addr:street`, it will do the correct thing with `addr:place`.

Closing as data issue.
